test/e2e/webdriver/driver.js

```diff
@@ -9,13 +9,23 @@ const cssToXPath = require('css-to-xpath');
  * @param {Object} element - Selenium Element
  * @returns {Object} modified Selenium Element
  */
-function wrapElementWithAPI(element) {
+function wrapElementWithAPI(element, driver) {
   element.press = (key) => element.sendKeys(key);
   element.fill = async (input) => {
     // The 'fill' method in playwright replaces existing input
     await element.clear();
     await element.sendKeys(input);
   };
+  element.waitForElementState = async (state, timeout) => {
+    switch (state) {
+      case 'hidden':
+        return await driver.wait(until.stalenessOf(element), timeout);
+      case 'visible':
+        return await driver.wait(until.elementIsVisible(element), timeout);
+      default:
+        return null;
```

Comment:
Should this throw? I'm having trouble imagining how this could be useful.

Reply:
Probably, yeah.